### PR TITLE
[Arch] Fix LGTM warnings

### PR DESCRIPTION
**Pull request by chennes**
A number of small fixes addressing complaints from the LGTM static analyzer.

### CODE REVIEW

**Comment by berndhahnebach** [comment]
FreeCAD CI-repository:
branch:
https://gitlab.com/berndhahnebach/FreeCAD/-/commits/PR_5069

pipeline for commit 78ecda36632408:
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines/378167853

status:
success

all pipelines for each branch: 
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines?scope=branches

File: src/Mod/Arch/ArchCommands.py [hunk 1]
```diff
@@ -635,7 +633,7 @@ def removeShape(objs,mark=True):
                     v1 = dims[0].multVec(v1)
                     v2 = dims[0].multVec(v2)
                     line = Draft.makeLine(v1,v2)
-                    wal = ArchWall.makeWall(line,width=width,height=dims[3],name=name)
+                    _ = ArchWall.makeWall(line,width=width,height=dims[3],name=name)
```
Comment (yorikvanhavre) [inline src/Mod/Arch/ArchCommands.py:636]: No actual need to catch the returned value I guess
```suggestion
                    ArchWall.makeWall(line,width=width,height=dims[3],name=name)
```

File: src/Mod/Arch/ArchCommands.py [hunk 2]
```diff
@@ -1336,14 +1334,14 @@ def printMessage( message ):
     if FreeCAD.GuiUp :
         if sys.version_info.major < 3:
             message = message.decode("utf8")
-        reply = QtGui.QMessageBox.information( None , "" , message )
+        _ = QtGui.QMessageBox.information( None , "" , message )
```
Comment (yorikvanhavre) [inline src/Mod/Arch/ArchCommands.py:1337]:
```suggestion
        QtGui.QMessageBox.information( None , "" , message )
```

File: src/Mod/Arch/ArchCommands.py [hunk 3]
```diff
@@ -1336,14 +1334,14 @@ def printMessage( message ):
...
 
 def printWarning( message ):
     FreeCAD.Console.PrintMessage( message )
     if FreeCAD.GuiUp :
         if sys.version_info.major < 3:
             message = message.decode("utf8")
-        reply = QtGui.QMessageBox.warning( None , "" , message )
+        _ = QtGui.QMessageBox.warning( None , "" , message )
```
Comment (yorikvanhavre) [inline src/Mod/Arch/ArchCommands.py:1344]:
```suggestion
        QtGui.QMessageBox.warning( None , "" , message )
```

File: src/Mod/Arch/ArchMaterial.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
 import FreeCAD
 if FreeCAD.GuiUp:
-    import FreeCADGui, Arch_rc, os
```
Comment (yorikvanhavre) [inline src/Mod/Arch/ArchMaterial.py:24]: Baware that importing the _rc file is needed to load the resources and be able to use any of the :/icons/ later on, in this case at L206:
`return ":/icons/Arch_Material_Group.svg"`

**Review by yorikvanhavre — CHANGES_REQUESTED** [review]
I think just the removal of the Arch_rc module needs to be fixed (or changed somehow). The rest are mere suggestions...

File: src/Mod/Arch/ArchReference.py [hunk 5]
```diff
@@ -183,14 +183,14 @@ def cleanShape(self,shapedata,obj,materials):
...
                     for soledge in solid.Edges:
-                        if found:
-                            break
                         if edge.hashCode() == soledge.hashCode():
                             found = True
                             break
+                    if found:
+                        break
+                if found:
+                    break
```
Comment (chennes) [inline src/Mod/Arch/ArchReference.py:193]: @yorikvanhavre can you double-check this change? It's not just a reformat, it is an algorithm change: the outermost for loop has an `else` clause, which only makes sense if there is a `break` statement that affects that outer loop. So I guessed at what that clause might be, but I'm not 100% sure of this one.

**Comment by freecadci** [comment]
FreeCAD CI-repository:
branch:
https://gitlab.com/berndhahnebach/FreeCAD/-/commits/PR_5069

pipeline for commit 06a0e1e923ca01:
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines/378948281

status:
success

all pipelines for each branch: 
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines?scope=branches

File: src/Mod/Arch/ArchReference.py (hunk 5 above)
Comment (yorikvanhavre) [reply src/Mod/Arch/ArchReference.py:193]: Yes, you are right... Indeed my implementation was not good, the else clause would almost never be triggered when it needs. I think yours fixes the problem

**Comment by yorikvanhavre** [comment]
I think this is good to go! I'll let you press the button unless you still want to add something

**Review by yorikvanhavre — APPROVED** [review]

**Comment by chennes** [comment]
Got it, thanks for reviewing!

**Pull request merged**